From 651e926ec27b139d288ed8a9c355d06444604512 Mon Sep 17 00:00:00 2001
From: Zhihao Cheng <chengzhihao1@huawei.com>
Date: Fri, 21 Oct 2022 16:01:07 +0800
Subject: [PATCH 048/595] [Huawei] kabi: net: fix kabi broken in sk_buff

Offering: HULK
hulk inclusion
category: bugfix
bugzilla: 187835
CVE: CVE-2022-2602

--------------------------------

In order to fix CVE-2022-2602, member 'scm_io_uring' is imported into
sk_buff, fix broken kabi in sk_buff.

Signed-off-by: Zhihao Cheng <chengzhihao1@huawei.com>
---
 include/linux/skbuff.h | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/include/linux/skbuff.h b/include/linux/skbuff.h
index e49fa09d9d30..26c431883c69 100644
--- a/include/linux/skbuff.h
+++ b/include/linux/skbuff.h
@@ -867,7 +867,6 @@ struct sk_buff {
 #ifdef CONFIG_TLS_DEVICE
 	__u8			decrypted:1;
 #endif
-	__u8			scm_io_uring:1;
 
 #ifdef CONFIG_NET_SCHED
 	__u16			tc_index;	/* traffic control index */
@@ -918,7 +917,7 @@ struct sk_buff {
 	__u32			headers_end[0];
 	/* public: */
 
-	KABI_RESERVE(1)
+	KABI_USE(1, __u8 scm_io_uring:1)
 	KABI_RESERVE(2)
 	KABI_RESERVE(3)
 	KABI_RESERVE(4)
-- 
2.17.1

